stan/math/rev/core/var.hpp

```diff
@@ -290,7 +290,108 @@ class var_value {
    * variable.
    */
-  inline var_value<T>& operator/=(T b);
+  inline var_value<T, VariType>& operator/=(T b);
+
+  /**
+   * A block view of the underlying Eigen matrices.
+   * @param i Starting row of block.
+   * @param j Starting columns of block.
+   * @param p Number of rows to return.
+   * @param q Number of columns to return.
+   */
+  inline const auto block(Eigen::Index i, Eigen::Index j, Eigen::Index p,
+                          Eigen::Index q) const {
+    using vari_sub = decltype(vi_->block(1, 1, 3, 3));
+    using var_sub = var_value<const typename vari_sub::PlainObject,
+                              const typename vari_sub::vari_type>;
+    return var_sub(new vari_sub(vi_->block(i, j, p, q)));
+  }
+
+  /**
+   * View of the head of Eigen vector types.
+   * @param n Number of elements to return from top of vector.
+   */
+  inline const auto head(Eigen::Index n) const {
+    using vari_sub = decltype(vi_->head(2));
+    using var_sub = var_value<const typename vari_sub::PlainObject,
+                              const typename vari_sub::vari_type>;
+    return var_sub(new vari_sub(vi_->head(n)));
+  }
+
+  /**
+   * View of the tail of the Eigen vector types.
+   * @param n Number of elements to return from bottom of vector.
+   */
+  inline const auto tail(Eigen::Index n) const {
+    using vari_sub = decltype(vi_->tail(2));
+    using var_sub = var_value<const typename vari_sub::PlainObject,
+                              const typename vari_sub::vari_type>;
+    return var_sub(new vari_sub(vi_->tail(n)));
+  }
+
+  /**
+   * View block of N elements starting at position `i`
+   * @param i Starting position of block.
+   * @param n Number of elements in block
+   */
+  inline const auto segment(Eigen::Index i, Eigen::Index n) const {
+    using vari_sub = decltype(vi_->segment(2, 3));
+    using var_sub = var_value<const typename vari_sub::PlainObject,
+                              const typename vari_sub::vari_type>;
+    return var_sub(new vari_sub(vi_->segment(i, n)));
+  }
+
+  /**
+   * View row of eigen matrices.
+   * @param i Row index to slice.
+   */
+  inline const auto row(Eigen::Index i) const {
+    using vari_sub = decltype(vi_->row(2));
+    using var_sub = var_value<const typename vari_sub::PlainObject,
+                              const typename vari_sub::vari_type>;
+    return var_sub(new vari_sub(vi_->row(i)));
+  }
+
+  /**
+   * View column of eigen matrices
+   * @param i Column index to slice
+   */
+  inline const auto col(Eigen::Index i) const {
+    using vari_sub = decltype(vi_->col(2));
+    using var_sub = var_value<const typename vari_sub::PlainObject,
+                              const typename vari_sub::vari_type>;
+    return var_sub(new vari_sub(vi_->col(i)));
+  }
+
+  /**
+   * View element of eigen matrices
+   * @param i Element to access
+   */
+  inline const auto operator()(Eigen::Index i) const {
+    using vari_coeff = decltype((*vi_)(2));
```

Comment:
[optional] I would prefere if you used actual arguments in these decltypes instead of some random values.
```suggestion
    using vari_coeff = decltype((*vi_)(i));
```